File: OOP/2_method.cpp
#include<bits/stdc++.h>
using namespace std;

class Car{
    public:
    string name;
    string engine;

    //constructor
    Car(string name, string engine){
        this->name = name;
        this->engine = engine;
        cout<<"This is a constructor"<<endl;
    }

    //method
    void display(){
        cout<<"Car is "<< name <<" and engine " <<engine<<endl;
    }

    //destructor
    ~Car(){
        cout<<"This is a Desctructor"<<endl;
    }
    // outside the class
  void car_fun();
};

// scope resolution operator
void Car::car_fun(){
  cout<<"This a car function"<<endl;
}


int main(){
  Car car("BMW","MOTOR");
  car.display();
  car.car_fun();
              
    return 0;
}

File: OOP/2_method.py

class Car:
    campany_name = "Volvo"  #class attributes
    def __init__(self,name,engine) -> None:
        #instance atributes
        self.name=name
        self.engine=engine
        print("This is  Constructor")
    
    def display(self):
        print("I this a method")

    def __del__(self):
        print("This is destructor")

    def __str__(self) -> str:
        return f'Company name {self.campany_name}'


car=Car('BMW',"Motorrr")
print(car)
car.price=5
print(car.__dict__)





